src/combatants.py

```diff
@@ -0,0 +1,56 @@
+import re
+from random import randint
+
+
+class DiceSimulator:
+    def encode(self, expr: str):
+        prefix_pattern = r'(\d+)d(\d+)'
+        suffix_pattern = r'\s+\+\s+(\d+)'
+        pattern = prefix_pattern + suffix_pattern
+        x = 0
+        y = 0
+        z = 0
+        if re.match(pattern, expr):
+            z = int(re.search(pattern, expr).group(3))
+        pattern = prefix_pattern
+        if re.match(pattern, expr):
+            x = re.search(pattern, expr).group(1)
+            y = re.search(pattern, expr).group(2)
+        self.__init(int(x), int(y), int(z))
+        return self
+
+    def __init(self, x: int, y: int, z: int):
```

Comment:
It would be better to move encoding dice information to a separate function and initialize this class with already extracted values.